src/room_buffer.rs

```diff
@@ -165,6 +163,19 @@ impl RoomBuffer {
         self.weechat_buffer().set_name(&name)
     }
 
+    /// Calculate the display name for a room member from its UserId. If no member with that ID is in
+    /// the room, the string representation of the ID will be returned.
+    fn calculate_user_name(&self, user_id: &UserId) -> String {
```

Comment:
This will need to implement display name disambiguation, e.g. when two users share the same display name we'll want to show the user id inside of brackets following the display name. Relevant spec entry can be found [here](https://matrix.org/docs/spec/client_server/latest#calculating-the-display-name-for-a-user).

Reply:
Should be fixed by 9efea35af099e139bce0957701a252a845bcb7a5

Reply:
That's going to be a bit expensive to calculate every time and it should live in the rust-sdk but it's good enough for now.

Reply:
Is it really? Getting from `HashMap` is `O(1)` and then `O(n)` for counting in the member list.
That's not the best but should perform OK.
Otherwise I think we need to hold a map from user ID to display name and update that on each join, quit, ...

Reply:
Yeah the counting is the problem, it's done on every received message for every room. Though it'll be a much smaller problem in Rust than it would have been in Python since we can yield back control to Weechat between every event/message.

A map that gets updated on a join/quit sounds good. Joins/quits aren't received as often as normal message events. We can leave it as is but a PR to the matrix-sdk with a better algorithm would be very much appreciated.